install.sh

```diff
@@ -60,22 +66,21 @@ if [ -e /etc/os-release ]; then
 
         GRUB_CFG_PATH='/boot/grub2/grub.cfg'
 
-        if [ -d /boot/efi/EFI/fedora ]
+        if [ -d /boot/efi/EFI/${ID} ]
         then
-            GRUB_CFG_PATH='/boot/efi/EFI/fedora/grub.cfg'
+            GRUB_CFG_PATH="/boot/efi/EFI/${ID}/grub.cfg"
         fi
 
         GRUB_DIR='grub2'
-        UPDATE_GRUB="grub2-mkconfig -o $GRUB_CFG_PATH"
+        UPDATE_GRUB="grub2-mkconfig -o ${GRUB_CFG_PATH}"
     fi
-fi
-
 
-echo 'Fetching theme archive'
-wget -O ${THEME}.zip https://github.com/shvchk/${THEME}/archive/master.zip
-
-echo 'Unpacking theme'
-unzip ${THEME}.zip
+    # BLS etries have 'kernel' class, copy corresponding icon
+    if [[ -d /boot/loader/entries && -e ${THEME}-master/icons/${ID}.png ]]
+    then
+        cp ${THEME}-master/icons/${ID}.png ${THEME}-master/icons/kernel.png
+    fi
```

Comment:
Is this Fedora/RH specific?

Reply:
I'm not aware of other distros, since I'm using Fedora, and they are using Bootloader Specification since version 30 (https://fedoraproject.org/wiki/Changes/BootLoaderSpecByDefault). Anyways, nothing is preventing other distros from introducing it in future releases.

Reply:
Ah, I was not aware of that, thanks. Still, `grub_class kernel` part seems to be Fedora/RH specific change to BLS (https://fedoraproject.org/wiki/Changes/BootLoaderSpecByDefault#Differences_from_BootLoaderSpec). Would you mind to move that code inside existing Fedora/RH `elif` block (lines 64..76)?

Reply:
Sure, no problem